rfcs/text/0000-application-monitoring-for-ga.md

```diff
@@ -0,0 +1,125 @@
+
+## Detailed Design
+
+In summary, each user-critical system in DCP should have the following.
+
+* Black-box availability monitoring using a health check
+* White-box monitoring of core system infrastructure
+* Alerts on black-box health check failures and white-box infrastructure anomalies
+* A dashboard that summarizes your white-box metrics
+
+The following subsections recommend concrete implementations of each requirement.
+
+### Overview of monitoring tooling
+
+Currently, the following tooling is in place to help DCP operators monitor their systems. Further, it indicates whether use of the tool should be adopted as a best practice within DCP as part of this proposal.
+
+| Purpose | Tool/framework | Deployments | Roadmap |
+|------|---------|-------------|--------------|
+| Centralized configuration for health checks, custom metrics, alerts, and dashboards | [dcp-monitoring](https://github.com/HumanCellAtlas/dcp-monitoring) | [github+terraform](https://github.com/HumanCellAtlas/dcp-monitoring) | adopt as best practice |
+| Visualization layer that can plot metrics from any datasource | Grafana | [dev/integration/staging](https://metrics.dev.data.humancellatlas.org/), [prod](https://metrics.data.humancellatlas.org/) | adopt as best practice |
+| Metrics for AWS applications and managed services | [CloudWatch Metrics](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#Metric) | [console](https://console.aws.amazon.com/cloudwatch/home?region=us-east-1#metricsV2:) | adopt as best practice for AWS |
+| Metrics for GCP applications and managed services | [Stackdriver Metrics](https://cloud.google.com/monitoring/api/metrics) | [console](https://console.cloud.google.com/monitoring) | adopt as best practice for GCP |
+| Alerting based on CloudWatch Metrics | [CloudWatch Alarms](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_concepts.html#CloudWatchAlarms) | [github+terraform](https://github.com/HumanCellAtlas/dcp-monitoring) | adopt as best practice |
+| Metrics based on operational log indices | [Elasticsearch Metrics](http://docs.grafana.org/features/datasources/elasticsearch/) | [dev/integration/staging](https://logs.dev.data.humancellatlas.org/), [prod](https://logs.data.humancellatlas.org/) | use not recommended, but sometimes necessary |
+| Build statuses, system health, and availability reporting | [Status API](https://github.com/HumanCellAtlas/status-api) | [dev/integration/staging](https://status.dev.data.humancellatlas.org/), [prod](https://status.data.humancellatlas.org/) | use optional |
+| Public presentation of build statuses, system health, and availability reporting | [DCP Status Page](https://github.com/HumanCellAtlas/humancellatlas.github.io) | [https://humancellatlas.github.io/](https://humancellatlas.github.io/) | temporary until formal status page is built
```

Comment:
who/which team is responsible for building and maintaining this formal status page? I haven't seen it mentioned in any charters. Does this make sense to integrate into the data browser?

Reply:
Good question. That's unclear at this point.